src/bin/commands/utils/init.ts

```diff
@@ -102,7 +105,7 @@ function createPackageJson(outputDirectory: string): void {
     dependencies: {
       "@aws-cdk/cloudformation-include": LibraryInfo.AWS_CDK_VERSION,
       "@aws-cdk/core": LibraryInfo.AWS_CDK_VERSION,
-      "@guardian/cdk": LibraryInfo.VERSION,
+      ...(!isTest && { "@guardian/cdk": LibraryInfo.VERSION }),
```

Comment:
Only add `@guardian/cdk` to `package.json` when in not in test as we will `npm link` it instead.

From https://docs.npmjs.com/cli/v8/commands/npm-link#caveat

> Note that package dependencies linked in this way are not saved to package.json by default, on the assumption that the intention is to have a link stand in for a regular non-link dependency. Otherwise, for example, if you depend on redis@^3.0.1, and ran npm link redis, it would replace the ^3.0.1 dependency with file:../path/to/node-redis, which you probably don't want! Additionally, other users or developers on your project would run into issues if they do not have their folders set up exactly the same as yours.
> 
> If you are adding a new dependency as a link, you should add it to the relevant metadata by running npm install <dep> --package-lock-only.
> 
> If you want to save the file: reference in your package.json and package-lock.json files, you can use npm link <dep> --save to do so.

This suggests we can `npm link @guardian/cdk --save`, however the `new` cli command will create `package.json` _and_ install deps in one step; we'd need to opt out of the installation in order to `npm link` at the right time as the order should be:
  - create `package.json`
  - mutate `package.json`
  - install deps

Reply:
The plot thickens... The ideal process would be:
  1. Generate `package.json` and other files, but don't install anything
  2. Mutate `package.json` to install `@guardian/cdk` from local clone
  3. Run `./script/ci` that was generated in 1

However, `./script/ci` fails with [linting errors](https://github.com/guardian/cdk/blob/f224a9517a8270c7107ea140ab131a28105e507b/src/bin/commands/bootstrap.ts#L138). That is, `./script/ci` will always fail.

Whilst the code change in this PR isn't ideal, I think any improvements would need a bigger change, in it's own PR.

Reply:
This is a neat solution; thanks for explaining it 👍

Reply:
The branch has been updated to `npm pack` and install from the resulting tarball as it is more representative of real world usage. The effect/need of this change is still the same however.